### ISSUE TITLE
Tailwind Based Redesign (enhancement?) 

### ISSUE THREAD

**Issue by kerns** [elleracompany/craft-cookie-consent#29]
Just offering to work with you on a new design implementation in tailwind that completely rethinks the way we interface with the javascript layer, ...with an eye toward making future upgrades (or other layouts) much, much easier to roll-out. So a more concrete/disciplined abstraction of function and layout, ...if it's possible.

This is still a little rough but it's from my very much working implementation. It's a bottom bar modal that also employs a full screen overlay to prevent interaction with the site until some form of consent – this could be an option.

<img width="972" alt="Screen Shot 2020-05-29 at 15 30 25" src="https://user-images.githubusercontent.com/20254/83266640-65fd1180-a1c3-11ea-9466-afae8e0785e7.png">

The tab is a small button at the bottom...

<img width="914" alt="Screen Shot 2020-05-29 at 15 36 36" src="https://user-images.githubusercontent.com/20254/83266847-aeb4ca80-a1c3-11ea-9bef-7f29b4c45c8c.png">

That opens this full screen with details and options for each group...

<img width="966" alt="Screen Shot 2020-05-29 at 15 35 19" src="https://user-images.githubusercontent.com/20254/83266657-6b5a5c00-a1c3-11ea-88f4-096821a8df37.png">

Maybe we can work on something more in this direction in the future. I know a lot of developers would be delighted to see something tailwind based.

**Comment by sieir** [comment]
This is definitely interesting and something we have thought about doing for a while. Would your implementation of this be something you would like to share with us?

**Comment by kerns** [comment]
Absolutely. Only, I would write this differently I think in a greenfield scenario. I was really just adapting to the code model already in place ...and I took some shortcuts. 😬 But, well ...just let me know how I can help.

**Comment by sieir** [comment]
My apologies for the slow response time. Totally understandable with the shortcuts, it would just be interesting and helpful to see your implementation to get an external perspective :) Do you have a fork of this or another preferred way for us to have a look?

**Comment by kerns** [comment]
No worries. I will think about the easiest way to hand-off in light of my schedule at the moment. I am also using some Sass/.scss for the generation of lightswitches... it could be nice that you include these in your dist as well. I will look at it as soon as I can!

**Comment by MichaelBrauner** [comment]
Yes. That would be very nice. Looking forward to it!

**Comment by kerns** [comment]
> Yes. That would be very nice. Looking forward to it!

Let's setup a playground some place using TW 2.0. I can rebuild the same or similar design, one that is also dark mode compatible.

**Comment by sieir** [comment]
> > Yes. That would be very nice. Looking forward to it!
> 
> Let's setup a playground some place using TW 2.0. I can rebuild the same or similar design, one that is also dark mode compatible.

Would perhaps a repository fork with a pull request suffice?
https://gist.github.com/Chaser324/ce0505fbed06b947d962

**Comment by kevinmu17** [comment]
This is a great idea to make available within the resource folder of the plugin (where script.js & style.css rest)
A developer can than choose to use a tailwind build or not at the site settings > template 

I would not recommend to make tailwind a default because you lose agility when hooking into just 1 framework

**Comment by jellingsen** [comment]
Thanks again for the offer and the screenshots — closing this as not planned in its original form. Since this was opened, the plugin gained full custom template support: point "Template" at your own Twig file and you control all markup/styling (Tailwind included), while the plugin handles consent storage and the JS endpoints. If you build a Tailwind-based template you'd like to share, we'd be happy to link it from the README.
